src/MedPrep.Api/Services/BunnyVideoService.cs

```diff
@@ -105,4 +110,70 @@ await this.teacherRepository.GetbyIdAsync(command.TeacherId)
             throw;
         }
     }
+
+    public async Task<VideoRequestResponse> FetchVideoInfo(Guid videoId)
+    {
+        //fetch the video Id from the database (video respositiory)
+        var video = await this.videoRepository.GetByIdAsync(videoId) ?? throw new NotFoundException("Video Not Found");
+
+        var nextVideo = video.NextVideo;
+        var prevVideo = video.PrevVideo;
+        var courseModule = video.CourseModule;
+
+        // Fetch video play data from bunny stream
+        var playData = await this.bunnyHttpClient.GetVideoPlayDataAsync(video.ThirdPartyVideoId);
+
+        // Map to video response Data to Object
+        var videoResponse = new VideoRequestResponse
+        {
+            Title = video.Title,
+            Description = video.Description,
+            NextVideo = nextVideo?.Id,
+            PrevVideo = prevVideo?.Id,
+            VideoSource = playData.VideoSources.Select(vs => $"video-{vs.SourceId}-{vs.Quality}").ToList(),
+            SubtitleSource = playData.Subtitles.Select(sub => $"subtitle-{sub.SourceId}-{sub.Language}").ToList(),
+            CourseModule = courseModule != null ? new CourseModuleDto
+            {
+                Name = courseModule.Topic,
+                Id = courseModule.Id
+
+            } : null!,
+            Playlist = video.Playlist != null ? new PlaylistDto
+            {
+                Name = video.Playlist.Name,
+                Id = video.Playlist.Id
+            } : null!
+
+
+        };
+
+        return videoResponse;
+    }
+    public async Task<VideoPlayResponse> PlayVideo(Guid videoId)
+    {
+        var client = new RestClient(BunnyStreamBaseUrl);
```

Comment:
Was there a reason to mix the http call implementation in the service? I just think is best to have it in the http client so that we don't constantly need to repeat the same calls in multiple methods.